agenta-backend/agenta_backend/services/evaluation_service.py

```diff
@@ -26,75 +33,165 @@ class UpdateEvaluationScenarioError(Exception):
     pass
 
 
-async def create_new_evaluation(
-    newEvaluationData: NewEvaluation, **kwargs: dict
-) -> Dict:
-    evaluation = newEvaluationData.dict()
-    evaluation["created_at"] = evaluation["updated_at"] = datetime.utcnow()
-    evaluation.update(kwargs)
+async def create_new_evaluation(payload: NewEvaluation, **kwargs: dict) -> Dict:
+    # Get user object
+    user = await get_user_object(kwargs["user_id"])
+
+    # Convert payload data to dictionary
+    evaluation_dict = payload.dict()
+    evaluation_dict["created_at"] = datetime.utcnow()
+    evaluation_dict["updated_at"] = datetime.utcnow()
+
+    # Initialize evaluation type settings embedded model
+    similarity_threshold = payload.evaluation_type_settings.similarity_threshold
+    evaluation_type_settings = EvaluationTypeSettings(
+        similarity_threshold=0.0
+        if similarity_threshold is None
+        else similarity_threshold
+    )
 
-    newEvaluation = await evaluations.insert_one(evaluation)
+    # Initialize evaluation instance and save to database
+    eval_instance = EvaluationDB(
+        status=payload.status,
+        evaluation_type=payload.evaluation_type,
+        evaluation_type_settings=evaluation_type_settings,
+        llm_app_prompt_template=payload.llm_app_prompt_template,
```

Comment:
why is the llm_app_prompt_template not part of the evaluation_type_settings?

Reply:
In the database schema shared by @aakrem, llm_app_prompt_template is an independent column in the evaluations table/schema.

Reply:
Alright!